### PR TITLE
spec: remove facet ref from core

### PR DESCRIPTION
**Pull request by JDarDagran**
### Problem

There are `$ref: facets/...` references in core `OpenLineage.json`. This breaks compatibility with JSON Schema specification.

### Solution

Remove references from core `OpenLineage.json`. Add spec tests.

#### One-line summary:

Remove references to facets in core spec.

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [x] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [x] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

**Comment by JDarDagran** [comment]
Files with `_after` prefix are the ones generated with changes in this PR:

```
$ diff src/main/java/io/openlineage/client/OpenLineage.java src/main/java/io/openlineage/client/OpenLineage_after.java 
875c875
<       this.schemaURL = URI.create("https://openlineage.io/spec/2-0-1/OpenLineage.json#/$defs/RunEvent");
---
>       this.schemaURL = URI.create("https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/RunEvent");
1947c1947
<       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-1/OpenLineage.json#/$defs/RunFacet");
---
>       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/RunFacet");
2096c2096
<       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-1/OpenLineage.json#/$defs/InputDatasetFacet");
---
>       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/InputDatasetFacet");
2184c2184
<       this.schemaURL = URI.create("https://openlineage.io/spec/2-0-1/OpenLineage.json#/$defs/JobEvent");
---
>       this.schemaURL = URI.create("https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/JobEvent");
3392c3392
<       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-1/OpenLineage.json#/$defs/DatasetFacet");
---
>       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/DatasetFacet");
4398c4398
<       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-1/OpenLineage.json#/$defs/JobFacet");
---
>       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/JobFacet");
5509c5509
<       this.schemaURL = URI.create("https://openlineage.io/spec/2-0-1/OpenLineage.json#/$defs/DatasetEvent");
---
>       this.schemaURL = URI.create("https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/DatasetEvent");
6365c6365
<       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-1/OpenLineage.json#/$defs/OutputDatasetFacet");
---
>       this._schemaURL = URI.create("https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/OutputDatasetFacet");
```

`$ diff src/main/java/io/openlineage/server/OpenLineage.java src/main/java/io/openlineage/server/OpenLineage_after.java` produces no differences.

**Review by mobuchowski — APPROVED** [review]
Looks good for me. As this changes spec, let's wait for additional pair of eyes before merging though.

File: spec/facets/ExtractionErrorRunFacet.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "$schema": "https://json-schema.org/draft/2020-12/schema",
-  "$id": "https://openlineage.io/spec/facets/1-0-0/ExtractionErrorRunFacet.json",
+  "$id": "https://openlineage.io/spec/facets/1-0-1/ExtractionErrorRunFacet.json",
```
Comment (julienledem) [inline spec/facets/ExtractionErrorRunFacet.json:3]: Maybe this should be 1-1-0
even if this was an oversight initially, it does make this change not backward compatible.
https://github.com/OpenLineage/OpenLineage/blob/main/spec/Versioning.md#mechanism
and following the link there: https://docs.snowplow.io/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver/
```
That's why we decided to introduce our own schema versioning notion: SchemaVer.
SchemaVer is defined as follows: MODEL-REVISION-ADDITION

MODEL when you make a breaking schema change which will prevent interaction with any historical data
REVISION when you introduce a schema change which may prevent interaction with some historical data
ADDITION when you make a schema change that is compatible with all historical data
```

File: spec/OpenLineage.json [hunk 2]
```diff
@@ -119,11 +121,7 @@
             {
               "type": "object",
               "additionalProperties": { "$ref": "#/$defs/RunFacet" }
-            },
-            { "$ref": "facets/ErrorMessageRunFacet.json" },
-            { "$ref": "facets/ExternalQueryRunFacet.json" },
-            { "$ref": "facets/NominalTimeRunFacet.json" },
-            { "$ref": "facets/ParentRunFacet.json" }
+            }
```
Comment (julienledem) [inline spec/OpenLineage.json:124]: LGTM

File: spec/facets/ProcessingEngineRunFacet.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "$schema": "https://json-schema.org/draft/2020-12/schema",
-  "$id": "https://openlineage.io/spec/facets/1-0-0/ProcessingEngineRunFacet.json",
+  "$id": "https://openlineage.io/spec/facets/1-0-1/ProcessingEngineRunFacet.json",
```
Comment (julienledem) [inline spec/facets/ProcessingEngineRunFacet.json:3]: same comment

**Review by julienledem — COMMENTED** [review]
2 minor comments, otherwise this looks great.

File: spec/facets/ExtractionErrorRunFacet.json (hunk 1 above)
Comment (JDarDagran) [reply spec/facets/ExtractionErrorRunFacet.json:3]: @julienledem  I fully do agree after reqading versioning guide. I'm changing both to 1-1-0

Comment (JDarDagran) [reply spec/facets/ExtractionErrorRunFacet.json]: Done.

File: spec/facets/ProcessingEngineRunFacet.json (hunk 3 above)
Comment (JDarDagran) [reply spec/facets/ProcessingEngineRunFacet.json]: Done.

**Pull request merged**